packages/core/src/browser/dialogs.ts

```diff
@@ -194,25 +194,24 @@ export abstract class AbstractDialog<T> extends BaseWidget {
         this.activeElement = undefined;
         super.close();
     }
-
     protected onUpdateRequest(msg: Message): void {
         super.onUpdateRequest(msg);
         this.validate();
     }
 
-    protected validate(): void {
+    protected async validate(): Promise<void> {
```

Comment:
please apply cancellation pattern here and in accept method (accept should have own cancellation source):
```ts
    protected validateCancellationSource = new CancellationTokenSource();
    protected async validate(): Promise<void> {
        if (!this.resolve) {
            return;
        }
        this.validateCancellationSource.cancel();
        this.validateCancellationSource = new CancellationTokenSource();
        const token = this.validateCancellationSource.token;
        const value = this.value;
        const error = await this.isValid(value, 'preview');
        if (token.isCancellationRequested) {
            return;
        }
        this.setErrorMessage(error);
    }
```

The issue is that if `validate` is triggered twice then only last should win. The same for accept.